testmodels/src/main/java/com/amplifyframework/testmodels/customprimarykey/Post.java

```diff
@@ -0,0 +1,214 @@
+package com.amplifyframework.testmodels.customprimarykey;
+
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.ModelPrimaryKey;
+import com.amplifyframework.core.model.annotations.BelongsTo;
+import com.amplifyframework.core.model.annotations.HasMany;
+import com.amplifyframework.core.model.annotations.Index;
+import com.amplifyframework.core.model.annotations.ModelConfig;
+import com.amplifyframework.core.model.annotations.ModelField;
+import com.amplifyframework.core.model.query.predicate.QueryField;
+import com.amplifyframework.core.model.temporal.Temporal;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import static com.amplifyframework.core.model.query.predicate.QueryField.field;
+
+/** This is an auto generated class representing the Post type in your schema. */
+@SuppressWarnings("all")
+@ModelConfig(pluralName = "Posts", type = Model.Type.USER, version = 1)
+@Index(name = "undefined", fields = {"id","title"})
+public final class Post implements Model {
+  public static final QueryField ID = field("Post", "id");
+  public static final QueryField TITLE = field("Post", "title");
+  public static final QueryField BLOG = field("Post", "blogPostsId");
+  private final @ModelField(targetType="ID", isRequired = true) String id;
+  private final @ModelField(targetType="String", isRequired = true) String title;
+  private final @ModelField(targetType="Blog") @BelongsTo(targetName = "blogPostsId", type = Blog.class) Blog blog;
```

Comment:
Thought. we would always have `targetNames` in version 1?

Reply:
Updated the class.